service/static/js/rest_api.js

```diff
@@ -0,0 +1,502 @@
+        };
+
+        $("#flash_message").empty();
+
+        let ajax = $.ajax({
+                type: "PUT",
+                url: `/inventory/${product_id}/${condition}`,
+                contentType: "application/json",
+                data: JSON.stringify(data)
+            })
+
+        ajax.done(function(res){
+            update_form_data(res)
+            flash_message("Successfully updated product ID " + product_id)
+        });
+
+        ajax.fail(function(res){
+            flash_message(res.responseJSON.message)
+        });
+
+    });
+
+    // ****************************************
+    // Retrieve an Inventory object
+    // ****************************************
+
+    $("#retrieve-btn").click(function () {
+
+        let product_id = $("#inventory_product_id_ret_del").val();
+        let condition = $("#inventory_condition_ret_del").val();
+
+        $("#flash_message").empty();
+
+        let ajax = $.ajax({
+            type: "GET",
+            url: `/inventory/${product_id}/${condition}`,
+            contentType: "application/json",
+            data: ''
+        })
+
+        ajax.done(function(res){
+            update_form_data(res)
+            flash_message("Successfully retrieved info of product ID " + product_id)
+        });
+
+        ajax.fail(function(res){
+            clear_form_data()
+            flash_message(res.responseJSON.message)
+        });
+
+    });
+
+    // ****************************************
+    // Delete an Inventory object
+    // ****************************************
+
+    $("#delete-btn").click(function () {
+
+        let product_id = $("#inventory_product_id_ret_del").val();
+        let condition = $("#inventory_condition_ret_del").val();
+
+        $("#flash_message").empty();
+
+        let ajax = $.ajax({
+            type: "DELETE",
+            url: `/inventory/${product_id}/${condition}`,
+            contentType: "application/json",
+            data: '',
+        })
+
+        ajax.done(function(res){
+            clear_form_data()
+            flash_message("Successfully deleted product ID " + product_id)
+        });
+
+        ajax.fail(function(res){
+            flash_message("Server error!")
+        });
+    });
+
+    // ****************************************
+    // Clear the form
+    // ****************************************
+
+    $("#clear-btn").click(function () {
+        $("#inventory_id").val("");
+        $("#flash_message").empty();
+        clear_form_data()
+    });
+
+    // ****************************************
+    // Build a list of items that need to be restocked
+    // ****************************************
+
+    $("#search-btn-restock").click(function () {
+        $("#flash_message").empty();
+
+        let ajax = $.ajax({
+            type: "GET",
+            url: `/inventory/RESTOCK`,
+            contentType: "application/json",
+            data: ''
+        })
+
+        ajax.done(function(res){         
```

Comment:
Feel free to make changes to this file in the future. There's a lot of boilerplate code in this method. I made a new function and copy/pasted all of the display code into it, but nothing happens when I call it from here